src/layer/rail.js

```diff
@@ -57,8 +57,12 @@ var defRail = {
 };
 
 var serviceSelector = ["match", ["get", "service"], ["siding", "spur", "yard"]];
-var isService = ["in", "service", "siding", "spur", "yard"];
-var isNotService = ["!in", "service", "siding", "spur", "yard"];
+var isService = [
+  "in",
+  ["get", "service"],
+  ["literal", ["siding", "spur", "yard"]],
```

Comment:
Use a `match` expression instead of searching for the value inside a literal array.